share/eds/cache/cache.go

```diff
@@ -38,6 +38,9 @@ type Cache interface {
 
 	// EnableMetrics enables metrics in Cache
 	EnableMetrics() error
+
+	// CloseMetrics closes metrics in Cache
+	CloseMetrics() error
```

Comment:
Modify Cache interface to return unreg func from EnableMetrics(). This will make unreg  to be available only when metrics are enabled.
```
EnableMetrics() (unreg func(), err error)
```

Reply:
done https://github.com/celestiaorg/celestia-node/pull/3281/commits/1c2188e5299497cb7053186afdc8eb80e931dd8b